File: htdocs/entity/EntityCorsa.php
<?php

class EntityCorsa {
    private int $idCorsa;
    private int $idTratta;
    private DateTime $data;
    private DateTime $orarioPartenza;
    private DateTime $orarioArrivo;
    private float $costo;
    private int $idAutobus;

    public function __construct(int $idCorsa, int $idTratta, DateTime $data, DateTime $orarioPartenza, DateTime $orarioArrivo, float $costo, int $idAutobus) {
        $this->idCorsa = $idCorsa;
        $this->idTratta = $idTratta;
        $this->data = $data;
        $this->orarioPartenza = $orarioPartenza;
        $this->orarioArrivo = $orarioArrivo;
        $this->costo = $costo;
        $this->idAutobus = $idAutobus;
    }

    // Getter e Setter per idCorsa
    public function getIdCorsa(): int {
        return $this->idCorsa;
    }

    public function setIdCorsa(int $idCorsa): void {
        $this->idCorsa = $idCorsa;
    }

    // Getter e Setter per idTratta
    public function getIdTratta(): int {
        return $this->idTratta;
    }

    public function setIdTratta(int $idTratta): void {
        $this->idTratta = $idTratta;
    }

    // Getter e Setter per data
    public function getData(): DateTime {
        return $this->data;
    }

    public function setData(DateTime $data): void {
        $this->data = $data;
    }

    // Getter e Setter per orarioPartenza
    public function getOrarioPartenza(): DateTime {
        return $this->orarioPartenza;
    }

    public function setOrarioPartenza(DateTime $orarioPartenza): void {
        $this->orarioPartenza = $orarioPartenza;
    }

    // Getter e Setter per orarioArrivo
    public function getOrarioArrivo(): DateTime {
        return $this->orarioArrivo;
    }

    public function setOrarioArrivo(DateTime $orarioArrivo): void {
        $this->orarioArrivo = $orarioArrivo;
    }

    // Getter e Setter per costo
    public function getCosto(): float {
        return $this->costo;
    }

    public function setCosto(float $costo): void {
        $this->costo = $costo;
    }

    // Getter e Setter per idAutobus
    public function getIdAutobus(): int {
        return $this->idAutobus;
    }

    public function setIdAutobus(int $idAutobus): void {
        $this->idAutobus = $idAutobus;
    }
}
?>

File: src/it/nominasuntsubstantiarerum/netbus/entity/EntityCorsa.java
package it.nominasuntsubstantiarerum.netbus.entity;

import java.time.LocalDate;

import it.nominasuntsubstantiarerum.netbus.util.Time;

public class EntityCorsa {
	private int idCorsa;
	private int idTratta;
	private LocalDate data;
	private Time orarioDiPartenza;
	private Time orarioDiArrivo;
	private float costo;
	private int idAutobus;
	
	public EntityCorsa(int idCorsa, int idTratta, LocalDate data, Time orarioDiPartenza, Time orarioDiArrivo, float costo, int idAutobus) {
		this.idCorsa = idCorsa;
		this.idTratta = idTratta;
		this.data = data;
		this.orarioDiPartenza = orarioDiPartenza;
		this.orarioDiArrivo = orarioDiArrivo;
		this.costo = costo;
		this.idAutobus = idAutobus;
	}
	
	public int getIdCorsa() {
		return idCorsa;
	}
	
	public void setIdCorsa(int idCorsa) {
		this.idCorsa = idCorsa;
	}
	
	public LocalDate getData() {
		return data;
	}
	
	public void setData(LocalDate data) {
		this.data = data;
	}

	public int getTratta() {
		return idTratta;
	}

	public void setTratta(int idTratta) {
		this.idTratta = idTratta;
	}

	public Time getOrarioDiPartenza() {
		return orarioDiPartenza;
	}

	public void setOrarioDiPartenza(Time orarioDiPartenza) {
		this.orarioDiPartenza = orarioDiPartenza;
	}

	public Time getOrarioDiArrivo() {
		return orarioDiArrivo;
	}

	public void setOrarioDiArrivo(Time orarioDiArrivo) {
		this.orarioDiArrivo = orarioDiArrivo;
	}

	public float getCosto() {
		return costo;
	}

	public void setCosto(float costo) {
		this.costo = costo;
	}
	
	public int getIdAutobus() {
		return idAutobus;
	}
	
	public void setIdAutobus(int idAutobus) {
		this.idAutobus = idAutobus;
	}
}
